### PR TITLE
fix: add error type for invalid configs

### PR DESCRIPTION
**Pull request by nagarajmanjunath**
## Description

This PR defines `ConfigError` type for internal modules.

## Related issue(s)
It partially fixes issue #315

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -13,21 +13,15 @@ import (
...
 	"github.com/pactus-project/pactus/www/nanomsg"
 	"github.com/pelletier/go-toml"
 )
 
-var (
-	//go:embed example_config.toml
-	exampleConfigBytes []byte
-
-	//go:embed bootstrap.json
-	bootstrapInfosBytes []byte
```
Comment (b00f) [inline config/config.go:29]: Why did you remove this?

File: config/config.go [hunk 2]
```diff
@@ -120,9 +102,21 @@ func DefaultConfigMainnet() *Config {
 	return conf
 }
 
+//nolint:lll // long multi-address
 func DefaultConfigTestnet() *Config {
 	conf := defaultConfig()
-	conf.Network.DefaultBootstrapAddrStrings = []string{}
+	conf.Network.DefaultBootstrapAddrStrings = []string{
```
Comment (b00f) [inline config/config.go:108]: I think your branch is too old. This is old Testnet.

**Review by b00f — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 1 above)
Comment (nagarajmanjunath) [reply config/config.go]: fixed !

File: config/errors.go [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package config
+
+// ConfigError is returned when the store configuration is invalid.
```
Comment (b00f) [inline config/errors.go:3]:
```suggestion
// ConfigError is returned when the node configuration is invalid.
```

File: txpool/errors.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package txpool
+
+// ConfigError is returned when the store configuration is invalid.
```
Comment (b00f) [inline txpool/errors.go:3]:
```suggestion
// ConfigError is returned when the txpool configuration is invalid.
```

**Review by b00f — APPROVED** [review]

**Comment by b00f** [comment]
@nagarajmanjunath please fix the linting issues

**Comment by b00f** [comment]
@nagarajmanjunath update the PR description

**Review by b00f — APPROVED** [review]

**Comment by b00f** [comment]
@nagarajmanjunath Merged with Thanks

**Pull request merged**